### ISSUE TITLE
IETF test cases - 1 of 64 FAIL (63 of 64 PASS) in frame.TestFrameHash (decoding audio samples)

### ISSUE THREAD

**Issue by mewmew** [mewkiz/flac#60]
This issue is a follow-up on #54 (which handles FLAC stream decoding). In particular, this issue handles issues of FLAC audio sample decoding (e.g. mismatch between expected audio samples and decoded audio samples).

The [IETF flac-test-files](https://github.com/ietf-wg-cellar/flac-test-files/tree/main/subset) repository contains 64 FLAC files, published by @ktmf01, which help exercise corner cases of the FLAC decoder.

When adding these test cases to the test suite, 7 of 64 FLAC files resulted in audio sample decoding failures, namely:

```
    --- FAIL: TestFrameHash/../testdata/flac-test-files/subset/16_-_partition_order_8_containing_escaped_partitions.flac (0.05s)
    --- FAIL: TestFrameHash/../testdata/flac-test-files/subset/22_-_12_bit_per_sample.flac (3.24s)
    --- FAIL: TestFrameHash/../testdata/flac-test-files/subset/32_-_high_resolution_audio,_partition_order_8_containing_escaped_partitions.flac (0.14s)
    --- FAIL: TestFrameHash/../testdata/flac-test-files/subset/37_-_20_bit_per_sample.flac (11.43s)
    --- FAIL: TestFrameHash/../testdata/flac-test-files/subset/62_-_predictor_overflow_check,_20-bit.flac (1.63s)
    --- FAIL: TestFrameHash/../testdata/flac-test-files/subset/63_-_predictor_overflow_check,_24-bit.flac (0.01s)
    --- FAIL: TestFrameHash/../testdata/flac-test-files/subset/64_-_rice_partitions_with_escape_code_zero.flac (0.01s)
```

This issue tracks the resolution of the identified decoding issues.

- [x] `16_-_partition_order_8_containing_escaped_partitions.flac`
- [x] `22_-_12_bit_per_sample.flac`
- [x] `32_-_high_resolution_audio,_partition_order_8_containing_escaped_partitions.flac`
- [x] `37_-_20_bit_per_sample.flac`
- [x] `62_-_predictor_overflow_check,_20-bit.flac`
- [x] `63_-_predictor_overflow_check,_24-bit.flac`
- [x] `64_-_rice_partitions_with_escape_code_zero.flac`

To reproduce, run:

```bash

go test -v github.com/mewkiz/flac/frame
```

Remember to initiate the submodule for IETF test cases in the [testdata/flac-test-files](https://github.com/ietf-wg-cellar/flac-test-files/tree/c3049d7f15491d256414adcf7da5e71782d09861) directory:
```bash
git submodule update --init --recursive
```

**Comment by mewmew** [comment]
MD5 mismatches for decoded audio samples:

```
=== FAIL   TestFrameHash/../testdata/flac-test-files/subset/16_-_partition_order_8_containing_escaped_partitions.flac
MD5 checksum mismatch for decoded audio samples; expected d0e1313950dc04b749c53cd349251bed, got 1d1380204d7eb5b138fe884178e9b67d

=== FAIL   TestFrameHash/../testdata/flac-test-files/subset/32_-_high_resolution_audio,_partition_order_8_containing_escaped_partitions.flac
MD5 checksum mismatch for decoded audio samples; expected 88c57117179b46e784643de9f20c1917, got b7377fd70228aed730cbb00e55dbafab

=== FAIL   TestFrameHash/../testdata/flac-test-files/subset/63_-_predictor_overflow_check,_24-bit.flac
MD5 checksum mismatch for decoded audio samples; expected e4e4a6b3a672a849a3e2157c11ad23c6, got a0343afaaaa6229266d78ccf3175eb8d

=== FAIL   TestFrameHash/../testdata/flac-test-files/subset/64_-_rice_partitions_with_escape_code_zero.flac
MD5 checksum mismatch for decoded audio samples; expected 0885019a14d23a6759404c96f525a9d4, got ef5f7a4d27bd066acf31c7a68d5758df
```

**Comment by mewmew** [comment]
`mewkiz/flac` audio samples for IETF test case `subset/16 - ...flac`
![test_case_16_go](https://github.com/mewkiz/flac/assets/1414531/9f5072d9-c4dd-42f4-ad10-ecb35dd64e92)

FLAC reference audio samples for IETF test case `subset/16 - ...flac`
![test_case_16_ref](https://github.com/mewkiz/flac/assets/1414531/bfac9fbf-3299-4153-9b1e-612febc01f89)

**Comment by mewmew** [comment]
The last remaining test case that's failing is `63 - predictor overflow check, 24-bit.flac`.

`mewkiz/flac` audio samples for IETF test case `subset/63 - ...flac`
![test_case_63_go](https://github.com/mewkiz/flac/assets/1414531/57266ae7-1fa5-46ab-ac8f-05995531de36)

FLAC reference audio samples for IETF test case `subset/63 - ...flac`
![test_case_63_ref](https://github.com/mewkiz/flac/assets/1414531/1b670a63-8758-4693-88da-de074294176f)

**Comment by mewmew** [comment]
Fixed by @MarkKremer in #61.

**Comment by mewmew** [comment]
(from: https://github.com/mewkiz/flac/pull/61#issuecomment-1778004889)

The v1.0.9 release has now been tagged.

Very happy to see that the audio samples of the IETF test cases are now correctly decoded by mewkiz/flac.

Cheerful regards,
Henry & Robin
